core/src/main/java/com/gentics/mesh/core/verticle/node/BinaryFieldHandler.java

```diff
@@ -227,57 +229,60 @@ public void handleUpdateBinaryField(InternalActionContext ac, String uuid, Strin
 				throw error(BAD_REQUEST, "error_found_field_is_not_binary", fieldName);
 			}
 
-			return db.tx(() -> {
-				SearchQueueBatch batch = searchQueue.create();
-				// Create a new node version field container to store the upload
-				NodeGraphFieldContainer newDraftVersion = node.createGraphFieldContainer(language, release, ac.getUser(), latestDraftVersion, true);
-				BinaryGraphField field = newDraftVersion.createBinary(fieldName);
-				String fieldUuid = field.getUuid();
-
-				Single<ImageInfo> obsImage;
-
-				// 3. Only gather image info for actual images. Otherwise return an empty image info object.
-				if (contentType.startsWith("image/")) {
-					try {
+			SearchQueueBatch batch = searchQueue.create();
+			// Create a new node version field container to store the upload
+			NodeGraphFieldContainer newDraftVersion = node.createGraphFieldContainer(language, release, ac.getUser(), latestDraftVersion, true);
+			BinaryGraphField field = newDraftVersion.createBinary(fieldName);
+			String fieldUuid = field.getUuid();
+
+			Single<ImageInfo> obsImage;
+
+			// 3. Only gather image info for actual images. Otherwise return an empty image info object.
+			if (contentType.startsWith("image/")) {
+				try {
+					ImageInfo imageInfo = ac.get("imageInfo");
+					if (imageInfo != null) {
+						obsImage = Single.just(imageInfo);
+					} else {
 						obsImage = imageManipulator.readImageInfo(() -> {
-							try {
-								return new FileInputStream(ul.uploadedFileName());
-							} catch (Exception e) {
-								log.error("Could not load schema for node {" + node.getUuid() + "}");
-								throw error(INTERNAL_SERVER_ERROR, "could not find upload file", e);
-							}
-						});
-					} catch (Exception e) {
-						log.error("Could not load schema for node {" + node.getUuid() + "}");
-						throw error(INTERNAL_SERVER_ERROR, "could not find upload file", e);
+						try {
+							return new FileInputStream(ul.uploadedFileName());
```

Comment:
Check why the stream is opened here and not closed - Why not use a try-with-resource here?

Reply:
This is already done in the readImageInfo method.